File: sdkwork-ai-sdk-swift/Sources/API/Images2Api.swift
import Foundation

public class Images2Api {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Create image variation
    public func createImageVariation(body: CreateImageVariationPostRequest? = nil, params: [String: Any]? = nil) async throws -> ImageGenerationResponse? {
        let response = try await client.post(ApiPaths.aiPath("/v1/images/variations"), body: body, params: params, headers: nil, contentType: "multipart/form-data")
        return response as? ImageGenerationResponse
    }

    /// Create image
    public func createImage(body: ImageGenerationRequest) async throws -> ImageGenerationResponse? {
        let response = try await client.post(ApiPaths.aiPath("/v1/images/generations"), body: body)
        return response as? ImageGenerationResponse
    }

    /// Create image edit
    public func createImageEdit(body: CreateImageEditRequest? = nil, params: [String: Any]? = nil) async throws -> ImageGenerationResponse? {
        let response = try await client.post(ApiPaths.aiPath("/v1/images/edits"), body: body, params: params, headers: nil, contentType: "multipart/form-data")
        return response as? ImageGenerationResponse
    }
}

File: sdkwork-ai-sdk-csharp/Api/Images2Api.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Ai.Http;
using Ai.Models;

namespace Ai.Api
{
    public class Images2Api
    {
        private readonly HttpClient _client;

        public Images2Api(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// Create image variation
        /// </summary>
        public async Task<ImageGenerationResponse?> CreateImageVariationAsync(CreateImageVariationPostRequest? body = null, Dictionary<string, object>? query = null)
        {
            return await _client.PostAsync<ImageGenerationResponse>(ApiPaths.AiPath("/v1/images/variations"), body, query, null, "multipart/form-data");
        }

        /// <summary>
        /// Create image
        /// </summary>
        public async Task<ImageGenerationResponse?> CreateImageAsync(ImageGenerationRequest body)
        {
            return await _client.PostAsync<ImageGenerationResponse>(ApiPaths.AiPath("/v1/images/generations"), body);
        }

        /// <summary>
        /// Create image edit
        /// </summary>
        public async Task<ImageGenerationResponse?> CreateImageEditAsync(CreateImageEditRequest? body = null, Dictionary<string, object>? query = null)
        {
            return await _client.PostAsync<ImageGenerationResponse>(ApiPaths.AiPath("/v1/images/edits"), body, query, null, "multipart/form-data");
        }
    }
}
